gobblin-runtime/src/main/java/gobblin/runtime/LimitingExtractorDecorator.java

```diff
@@ -47,11 +57,15 @@
 
   private final Extractor<S, D> extractor;
   private final Limiter limiter;
-
-  public LimitingExtractorDecorator(Extractor<S, D> extractor, Limiter limiter) {
+  private final TaskState taskState;
+  private static final String LIMITER_PERMIT_ACQUIRE_FAILURE = "limiter.permit.acquire.failure";
```

Comment:
Usually the event names are camel cased. Also, try to use names that convey the practical meaning instead of the technical cause. For example `PrematureExtractorStop` (under metadata, you could include a `cause` field that says `LimiterPermitAcquireFailure`).